### ISSUE TITLE
RandomForest model size too large

### ISSUE THREAD

**Issue by xieleo5** [automl/auto-sklearn#1647]
## Describe the bug ##
I'm trying to use askl to solve problems on openml. When the dataset is large, the random forest model on the disk will be too large than expected. For a task with around 10k instance, the RF model can reach 6.5 GB. Is there any way to restrict the size of RF? 
Here is a screenshot of the model on disk:
![image](https://user-images.githubusercontent.com/55373943/224568599-3d756377-8146-4b41-865c-2eb98210b1a0.png)

Also, I tried with the RF model, it apparently overfit the dataset by a large amount. The training accuracy is 0.998 but the testing only get 0.67. 

## To Reproduce ##
```python
import openml
from autosklearn.classification import AutoSklearnClassifier

task = openml.tasks.get_task(189355)
X, y = task.get_X_and_y("dataframe")
X = X.sample(frac=1/41)
y = y[X.index]

X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(
    X, y, random_state=42
)

askl_config = {
    "time_left_for_this_task": 600,
    "tmp_folder": <PATH>,
    "delete_tmp_folder_after_terminate": False,
    "memory_limit": None,
}

automl = AutoSklearnClassifier(**askl_config)
automl.fit(X_train, y_train)
```

## Expected behavior ##
The model size on disk should be smaller, or have some way to limit the range of RandomForest's parameter.

**Comment by eddiebergman** [comment]
Hi @xieleo5, in general you can limit memory by setting the `"memory_limit"` but I guess from your code you've explicitly set it to `None`. Another option is use a different [`resampling_strategy`](https://automl.github.io/auto-sklearn/master/api.html#autosklearn.classification.AutoSklearnClassifier) that is less prone to overfitting.
